### PR TITLE
Populate CFG node types

### PR DESCRIPTION
**Pull request by AHGIJMKLKKZNPJKQR**

### CODE REVIEW

File: src/main/kotlin/cacophony/controlflow/CFGNode.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        val lhs: CFGNode,
+        val rhs: CFGNode,
+    ) : LogicalOperator
+
+    sealed class NotEquals(
+        val lhs: CFGNode,
+        val rhs: CFGNode,
+    ) : LogicalOperator
+
+    sealed class Lesser(
```
Comment (Rafi0221) [inline src/main/kotlin/cacophony/controlflow/CFGNode.kt:133]: Less? same below

File: src/main/kotlin/cacophony/controlflow/CFGNode.kt [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        val destination: CFGNode,
+    ) : Unconditional
+
+    sealed class Constant(
+        val value: Int,
+    ) : Unconditional,
+        Leaf
+
+    sealed class Sequence(
+        vararg val nodes: CFGNode,
```
Comment (KacperTopolski) [inline src/main/kotlin/cacophony/controlflow/CFGNode.kt:65]: I think we should avoid varargs in primary constructors and simply use lists. Maybe add it as a secondary contructor.

**Review by Rafi0221 — CHANGES_REQUESTED** [review]
Please add push and pop nodes.

File: src/main/kotlin/cacophony/controlflow/CFGNode.kt [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    sealed class Call :
+        Unconditional,
+        Leaf
+
+    // NOTE: Push may be unncessary since it can be done via Assignment + MemoryAccess
+    sealed class Push(
+        val value: CFGNode,
+    ) : Unconditional
+
+    // NOTE: Pop may be unncessary since it can be done via Assignment
```
Comment (KacperTopolski) [inline src/main/kotlin/cacophony/controlflow/CFGNode.kt:52]: Typo.

File: src/main/kotlin/cacophony/controlflow/CFGNode.kt [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        val lhs: CFGNode,
+        val rhs: CFGNode,
+    ) : ArithmeticOperator
+
+    sealed class Modulo(
+        val lhs: CFGNode,
+        val rhs: CFGNode,
+    ) : ArithmeticOperator
+
+    sealed interface BitwiseOperator : Unconditional
```
Comment (KacperTopolski) [inline src/main/kotlin/cacophony/controlflow/CFGNode.kt:109]: We do not support bitwise operators.

**Review by Rafi0221 — APPROVED** [review]

**Pull request merged**